File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {

	public void solution() throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		Stack<Integer> stack = new Stack<>();
		StringBuilder sb = new StringBuilder("");
		
		int N = Integer.valueOf(br.readLine());

		for (int i = 0; i < N; i++) {
			String command = br.readLine();

			if (command.startsWith("push")) {
				String[] temp = command.split(" ");
				command = temp[0];
				int number = Integer.valueOf(temp[1]);
				stack.add(number);
				continue;
			}

			if (command.equals("pop")) {
				if(stack.size() == 0) {
					sb.append(-1 + "\n");
					continue;
				} else {
					sb.append(stack.pop() + "\n");
					continue;
				}
			}

			if (command.equals("size")) {
				sb.append(stack.size() + "\n");
				continue;
			}

			if (command.equals("empty")) {
				if(stack.empty()) {
					sb.append(1 + "\n");
				} else {
					sb.append(0 + "\n");
				}
				continue;
			}

			if (command.equals("top")) {
				if(stack.empty()) {
					sb.append(-1 + "\n");
				}else {
					sb.append(stack.peek() + "\n");
				}
			}
		}
		System.out.println(sb);
	}

	public static void main(String[] args) throws Exception {
		new Main().solution();
	}
}

File: 백준/Silver/10828. 스택/스택.js
const fs = require('fs');
const filePath = '/dev/stdin';

const input = fs.readFileSync(filePath).toString().trim().split('\n');

const solution = input => {
  const [N, ...commands] = input;

  const stack = [];
  let pointer = 0;

  let answer = '';

  for (let i = 0; i < N; i++) {
    const command = commands[i].split(' ')[0];
    const number = commands[i].split(' ')[1];
    let result = '';

    switch (command) {
      case 'pop':
        if (pointer) {
          pointer--;
          result = stack.splice(-1); // 원본배열 보존 x
          answer += result + '\n';
        } else {
          answer += '-1' + '\n';
        }
        break;

      case 'push':
        stack[pointer++] = number;
        break;

      case 'size':
        result = pointer;
        answer += result + '\n';
        break;

      case 'empty':
        result = pointer ? 0 : 1;
        answer += result + '\n';
        break;

      case 'top':
        result = pointer ? stack[pointer - 1] : -1;
        answer += result + '\n';
        break;
      default:
        break;
    }
  }
  console.log(answer.trim());
};

solution(input);